common/src/main/java/com/microsoft/identity/common/internal/msafederation/google/SignInWithGoogleCredential.kt

```diff
@@ -22,21 +22,24 @@
 // THE SOFTWARE.
 package com.microsoft.identity.common.internal.msafederation.google
 
-import com.microsoft.identity.common.internal.msafederation.MsaFederationConstants.MSA_ID_TOKEN_HEADER_KEY
+import com.google.gson.annotations.SerializedName
 import com.microsoft.identity.common.internal.msafederation.FederatedCredential
 import com.microsoft.identity.common.internal.msafederation.FederatedSignInProviderName
+import com.microsoft.identity.common.internal.msafederation.MsaFederationConstants.MSA_ID_TOKEN_HEADER_KEY
+import java.io.Serializable
 
 /**
  * Represents credential artifact as result of successful sign in with google
  * It can contain id token.
  */
-data class SignInWithGoogleCredential internal constructor(internal val idToken: String) : FederatedCredential(FederatedSignInProviderName.GOOGLE) {
+data class SignInWithGoogleCredential internal constructor(
+    @SerializedName("idToken") internal val idToken: String) : FederatedCredential(FederatedSignInProviderName.GOOGLE), Serializable {
 
     /**
      * Helper method to create header that can be used in MSA
      * authorization. (Move to better place.)
      */
-    fun asHeaders(): Map<String, String> {
+    fun getIdProviderHeaders(): Map<String, String> {
```

Comment:
I suggest sticking to either abbreviation or full form word. `IdProvider` is somwhere in between.

I suggest choosing either `IdP` (a true abbreviator) or rather just using full word `IdentityProvider`

Reply:
Also, tbh, I can't really tell what this function is supposed to do by just looking at the method name. I mean it's clear that it is going to return a map that contains some headers but not clear on who/where/when these headers should be used. 

For example, someone may just get these and start attaching these to DRS Discovery requests...which is not what they are for.

Reply:
Refactored into extension methods.

named IdProvider keeping on authorization endpoint parameter "id_provider" as reference.